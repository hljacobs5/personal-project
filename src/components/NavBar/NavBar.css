button {
	display: flex;
	border-radius: 50%;
	cursor: pointer;
	width: 80px;
	height: 50px; 
	display: inline-block;
	justify-content: space-between;
	font-family: 'Kodchasan', sans-serif;
}

button:hover {
	background-color: #407888;
	color: white;
}

