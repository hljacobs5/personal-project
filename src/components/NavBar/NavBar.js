import React from 'react';
import PropTypes from 'prop-types';
import './NavBar.css';
import { Route, NavLink } from 'react-router-dom';

const NavBar = ({ fetchCategory }) => {
	return (
		<div>
			<NavLink
				name='politics'
				to='/politics'
				className='politics'>
				<button
					onClick={fetchCategory}
					name='politics'
					id='24'>Politics
				</button>
			</NavLink>
		</div>
	)
}

export default NavBar;