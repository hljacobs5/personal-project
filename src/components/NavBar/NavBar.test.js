import React from 'react';
import { shallow, mount } from 'enzyme';

import NavBar from './NavBar.js';

describe('NavBar', () => {
	let wrapper;
	let mockFunction;
	let mockFetchCategory;
	beforeEach(() => {
		mockFunction = jest.fn()
		wrapper = shallow(<NavBar
			mockFetchCategory={mockFunction}
		>);
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should call fetchCategory when politics is clicked', () => {
		wrapper.find('.politics').simulate('click')
		wrapper.instance().fetchCategory = mockFetchCategory
		expect(mockFunction).toBeCalled()
	});

	it('should call fetchCategory when animals is clicked', () => {
		wrapper.find('.animals').simulate('click')
		wrapper.instance().fetchCategory = mockFetchCategory
		expect(mockFunction).toBeCalled()
	});

	it('should call fetchCategory when environment is clicked', () => {
		wrapper.find('.environment').simulate('click')
		wrapper.instance().fetchCategory = mockFetchCategory
		expect(mockFunction).toBeCalled()
	});
});