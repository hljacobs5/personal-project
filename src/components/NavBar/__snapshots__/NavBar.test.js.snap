// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    fetchCategory={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="politics"
          name="politics"
          to="/politics"
        >
          <button
            id="24"
            name="politics"
            onClick={[MockFunction]}
          >
            Politics
          </button>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="animals"
          name="animals"
          to="/animals"
        >
          <button
            id="27"
            name="animals"
            onClick={[MockFunction]}
          >
            Animals
          </button>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="environment"
          name="environment"
          to="/environment"
        >
          <button
            id="17"
            name="animals"
            onClick={[MockFunction]}
          >
            Environment
          </button>
        </NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            id="24"
            name="politics"
            onClick={[MockFunction]}
          >
            Politics
          </button>,
          "className": "politics",
          "name": "politics",
          "to": "/politics",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Politics",
            "id": "24",
            "name": "politics",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Politics",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            id="27"
            name="animals"
            onClick={[MockFunction]}
          >
            Animals
          </button>,
          "className": "animals",
          "name": "animals",
          "to": "/animals",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Animals",
            "id": "27",
            "name": "animals",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Animals",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            id="17"
            name="animals"
            onClick={[MockFunction]}
          >
            Environment
          </button>,
          "className": "environment",
          "name": "environment",
          "to": "/environment",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Environment",
            "id": "17",
            "name": "animals",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Environment",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="politics"
            name="politics"
            to="/politics"
          >
            <button
              id="24"
              name="politics"
              onClick={[MockFunction]}
            >
              Politics
            </button>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="animals"
            name="animals"
            to="/animals"
          >
            <button
              id="27"
              name="animals"
              onClick={[MockFunction]}
            >
              Animals
            </button>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="environment"
            name="environment"
            to="/environment"
          >
            <button
              id="17"
              name="animals"
              onClick={[MockFunction]}
            >
              Environment
            </button>
          </NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              id="24"
              name="politics"
              onClick={[MockFunction]}
            >
              Politics
            </button>,
            "className": "politics",
            "name": "politics",
            "to": "/politics",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Politics",
              "id": "24",
              "name": "politics",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Politics",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              id="27"
              name="animals"
              onClick={[MockFunction]}
            >
              Animals
            </button>,
            "className": "animals",
            "name": "animals",
            "to": "/animals",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Animals",
              "id": "27",
              "name": "animals",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Animals",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              id="17"
              name="animals"
              onClick={[MockFunction]}
            >
              Environment
            </button>,
            "className": "environment",
            "name": "environment",
            "to": "/environment",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Environment",
              "id": "17",
              "name": "animals",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Environment",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
